src/core/lib/pick-layers.js

```diff
@@ -348,42 +348,53 @@ function getClosestFromPickingBuffer(gl, {
   deviceRect
 }) {
   assert(pickedColors);
-  let closestResultToCenter = NO_PICKED_OBJECT;
 
   // Traverse all pixels in picking results and find the one closest to the supplied
   // [deviceX, deviceY]
+  const {x, y, width, height} = deviceRect;
   let minSquareDistanceToCenter = deviceRadius * deviceRadius;
+  let closestPixelIndex = -1;
   let i = 0;
 
-  for (let row = 0; row < deviceRect.height; row++) {
-    for (let col = 0; col < deviceRect.width; col++) {
-      // Decode picked layer from color
-      const pickedLayerIndex = pickedColors[i + 3] - 1;
-
-      if (pickedLayerIndex >= 0) {
-        const dx = col + deviceRect.x - deviceX;
-        const dy = row + deviceRect.y - deviceY;
-        const d2 = dx * dx + dy * dy;
-
-        if (d2 <= minSquareDistanceToCenter) {
-          minSquareDistanceToCenter = d2;
-
-          // Decode picked object index from color
-          const pickedColor = pickedColors.slice(i, i + 4);
-          const pickedLayer = layers[pickedLayerIndex];
-          if (pickedLayer) {
-            const pickedObjectIndex = pickedLayer.decodePickingColor(pickedColor);
-            closestResultToCenter = {pickedColor, pickedLayer, pickedObjectIndex};
-          } else {
-            log.error(0, 'Picked non-existent layer. Is picking buffer corrupt?');
+  for (let row = 0; row < height; row++) {
```

Comment:
Would it make sense to search rows center out instead of from top to bottom? Might improve efficiency of row skipping.

Reply:
It's going to make the code a lot harder to read, as the pointer is not necessarily centered in `deviceRect` (edge clipping).